### DISCUSSION TITLE
Adding support for tailwind-theming or similar plugins

### DISCUSSION THREAD

**Discussion by irg1008** [ben-rogerson/twin.macro#476] [Ideas]
Twin.Macro is an awesome tool. I love using it with styled components.
The thing is in my last app I wanted to implement several themes and using the variant for each one gets very complex and doesn`t scale very well. The solution for this are of course theming plugins, wich Twin.Macro does not support yet.

Are they thought to be supported in the future?

Thanks!

**Comment by drosi94** [comment]
DaisyUI would be nice to be supported as well

**Reply by ben-rogerson** [reply]
> Everything seems to work pretty nice with the DaisyUI plugin (autocompletion, the utility/components classes are working as expected) but it takes really long time to build the application

When you say daisyui is unsupported do you mean to improve build speeds? 
Or is it more about having all daisyui components 100% tested as working with Twin?

**Comment by ben-rogerson** [comment]
> ... theming plugins ... Are they thought to be supported in the future?

Possibly - I've previously run into issues supporting the `addVariant` feature within tailwindcss plugins.
This will probably require a unique implementation which means plugins using `addVariant` will not work.
[Discussed here](https://github.com/ben-rogerson/twin.macro/discussions/320#discussioncomment-802591)

**Comment by ben-rogerson** [comment]
Hey folks, some good news - daisyui will work with the next version of twin (`rc-2`) 👍
